src/Microsoft.AspNetCore.WebSockets/WebSocketMiddleware.cs

```diff
@@ -42,7 +45,24 @@ public Task Invoke(HttpContext context)
             var upgradeFeature = context.Features.Get<IHttpUpgradeFeature>();
             if (upgradeFeature != null && context.Features.Get<IHttpWebSocketFeature>() == null)
             {
-                context.Features.Set<IHttpWebSocketFeature>(new UpgradeHandshake(context, upgradeFeature, _options));
+                var webSocketFeature = new UpgradeHandshake(context, upgradeFeature, _options);
+                context.Features.Set<IHttpWebSocketFeature>(webSocketFeature);
+
+                if (_options.AllowedOrigins.Count > 0)
```

Comment:
Should we support a `null` value here? And treat it the same as an empty collection? Just thinking defensively.

Reply:
Either guard against null or make the setter private.